spec/lib/mastodon/cli/accounts_spec.rb

```diff
@@ -662,4 +662,78 @@
       end
     end
   end
+
+  describe '#rotate' do
+    context 'when neither username nor --all option are given' do
+      it 'exits with an error message' do
+        expect { cli.rotate }.to output(
+          a_string_including('No account(s) given')
+        ).to_stdout
+          .and raise_error(SystemExit)
+      end
+    end
+
+    context 'when a username is given' do
+      let(:account) { Fabricate(:account) }
+
+      it 'correctly rotates keys for the specified account' do
+        old_private_key = account.private_key
+        old_public_key = account.public_key
+
+        cli.rotate(account.username)
+        account.reload
+
+        expect(account.private_key).to_not eq(old_private_key)
+        expect(account.public_key).to_not eq(old_public_key)
+      end
+
+      it 'broadcasts the new keys for the specified account' do
+        allow(ActivityPub::UpdateDistributionWorker).to receive(:perform_in)
+
+        cli.rotate(account.username)
+
+        expect(ActivityPub::UpdateDistributionWorker).to have_received(:perform_in).with(anything, account.id, anything).once
```

Comment:
Though this is handled by a private method (which I think shouldn't be tested), I found it relevant enough to be tested. Also, I'm not sure if it's a good way to write this expectation.

Reply:
Sounds fine to me!